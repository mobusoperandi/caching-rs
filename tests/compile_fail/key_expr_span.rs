use michie::memoized;
use std::collections::HashMap;

#[memoized(key_expr = b, store_type = HashMap<bool, bool>)]
fn f(a: bool) -> bool {
    a
}

fn main() {}
