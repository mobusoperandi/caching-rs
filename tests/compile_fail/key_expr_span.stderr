error[E0425]: cannot find value `b` in this scope
 --> tests/compile_fail/key_expr_span.rs:4:23
  |
4 | #[memoized(key_expr = b, store_type = HashMap<bool, bool>)]
  |                       ^ help: a local variable with a similar name exists: `a`
